v1/lib/v1/search_engine/river.rb

```diff
@@ -165,29 +187,75 @@ def self.verify_river_status(name=river_name)
-          else
-            index = service_meta(name)['_source']['index']['index']
-            return "River '#{name}' pointed at index '#{index}' running on node '#{node}'"
-          end
+          index = service_meta(name)['_source']['index']['index']
+          last_seq = last_sequence(name)
+          return "River '#{name}' pointed at index '#{index}' last sequence " \
+                 "#{last_seq} running on node '#{node}'"
         end
       end
 
+      ##
+      # Get the last sequence number from the River
+      #
+      # This method used to return nil if HTTParty.get failed. There were no
+      # unit tests, so we have to determine the correct behavior through
+      # analysis. `.recreate_river' above is now stricter about only
+      # _re_creating rivers, so maybe this function used to ignore errors
+      # because it was convenient to create a new river if one didn't exist.
+      #
+      # An exception will be raised if the River exists, but never started
+      # flowing because the source CouchDB database is still empty.  In that
+      # case, the HTTP GET request to the River endpoint will return HTTP
+      # status 404 and the response body will be a JSON object without
+      # the '_source' property. The remedy to this is to require a last
+      # sequence number when creating a River, which we now do, to eliminate
+      # ambiguity.  A last_seq property of "0" can be specified in that case.
+      #
+      # @return [Fixnum]       Sequence number
+      # @param  name [String]  Name of the River
+      # @raise  [RuntimeError] if the HTTP request to the River endpoint fails
+      # @see self.recreate_river
+      #
       def self.last_sequence(name=river_name)
         # name will be present but nil when called from a rake task
         name ||= river_name
-        response = HTTParty.get(endpoint(name) + '/_seq').parsed_response
-        seq = response['_source']['couchdb']['last_seq'] rescue nil
-
-        return nil if seq.nil?
+        response = HTTParty.get(endpoint(name) + '/_seq')
+        raise "Problem getting last sequence: #{JSON.parse(response.body)}" \
+          unless response.success?
+        seq = response.parsed_response['_source']['couchdb']['last_seq']
 
         # handle bigcouch's last_seq format if it looks like it
-        seq = JSON.parse(seq).first if seq !~ /^\d+$/
+        seq = JSON.parse(seq).first if seq !~ /^\d+$/  # but how not?  -mb
 
         seq.to_i
       end
 
+      ##
+      # Set the last sequence number for a River.
+      #
+      # The sequence number can be PUT for a River before the rest of it is
+      # created by a PUT to its _meta resource. Putting last_seq creates a data
+      # structure for the incomplete river. When you do this (before PUTing to
+      # `_meta') you'll see a property for the new river in `/_river/_mapping'.
+      #
+      # @param  seq_num [Fixnum or String] Sequence number /^\d+$/
```

Comment:
So, BigCouch reports a "sequence number" as a JSON array; first item is equivalent to the CouchDB last_seq, and the second is a long, opaque string. Upon river creation, the full array format is necessary, otherwise the call to BigCouch will fail: 

```
$ curl -sv "repo-stg1:5984/dpla/_changes?since=1"
* Connected to repo-stg1 port 5984 (#0)
> GET /dpla/_changes?since=1 HTTP/1.1
> Host: repo-stg1:5984
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< X-Couch-Request-ID: 3254ddde
< Server: CouchDB/1.1.1 (Erlang OTP/R14B01)
< Date: Thu, 29 Sep 2016 16:14:20 GMT
< Content-Type: text/plain;charset=utf-8
< Content-Length: 164
< Cache-Control: must-revalidate
<
{"error":"function_clause","reason":null,"stack":["chttpd_db:changes_callback/2","chttpd:handle_request/1","mochiweb_http:headers/5","proc_lib:init_p_do_apply/3"]}
```